### PR TITLE
Update device selectors to use hooks instead of local state

### PR DESCRIPTION
**Pull request by michhyun1**
**Issue #:** 

**Description of changes:**
Use hooks instead of a local state inside Device Input

**Testing**
1. Have you successfully run `npm run build:release` locally? yes

2. How did you test these changes?
- Selected difference devices in the /devices page (preview page) and verified that the correct selected devices showed up

3. If you made changes to the component library, have you provided corresponding documentation changes?
yes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+- Update Device selector to use hooks instead of local state
```
[the remark this answers is not in the document]
Comment (michhyun1) [reply CHANGELOG.md:8]: This is technically not supposed to be a part of the 1.2.0 release, so I just added it at the top and once the release is finished, then I'll create a new Unreleased section and add it to that section

File: src/components/sdk/DeviceSelection/DeviceInput.tsx [hunk 2]
```diff
@@ -60,7 +51,7 @@ const DeviceInput: React.FC<Props> = ({
       field={Select}
       options={options}
       onChange={selectDevice}
-      value={selectedDevice}
+      value={selectedDevice || ''}
```
[the remark this answers is not in the document]
Comment (michhyun1) [reply src/components/sdk/DeviceSelection/DeviceInput.tsx:54]: It looks like useEffect only setsSelectedDevice if there are any devices or selected devices. In the case that the deviceInput is rendered but there are no devices to choose from the selectedDevice state should still be  `' '`, which is the initial state of selectedDevices

File: src/components/sdk/DeviceSelection/DeviceInput.tsx [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
 import { FormField } from '../../ui/FormField';
 import { Select } from '../../ui/Select';
-import { DeviceType } from '../../../types';
+import { DeviceType, SelectedDeviceId } from '../../../types';
 
 interface Props {
   label: string;
   notFoundMsg: string;
   devices: DeviceType[];
+  selectedDevice: SelectedDeviceId;
```
[the remark this answers is not in the document]
Comment (michhyun1) [reply src/components/sdk/DeviceSelection/DeviceInput.tsx]: done! thanks

File: src/components/sdk/DeviceSelection/docs/SpeakerSelection.stories.mdx [hunk 4]
```diff
@@ -5,7 +5,7 @@ import SpeakerSelection from '../SpeakerSelection';
 
 # SpeakerSelection
 
-The `SpeakerSelection` component renders a `Select FormField` with audio input source options provided by the `useAudioOutputs` hook.
+The `SpeakerSelection` component renders a `Select FormField` with audio output source options and the selected audio output provided by the `useAudioOutputs` hook.
```
Comment (ziyiz-amzn) [inline src/components/sdk/DeviceSelection/docs/SpeakerSelection.stories.mdx:8]: `selected audio output device` ?

File: src/components/sdk/DeviceSelection/DeviceInput.tsx [hunk 5]
```diff
@@ -60,7 +51,7 @@ const DeviceInput: React.FC<Props> = ({
       field={Select}
       options={options}
       onChange={selectDevice}
-      value={selectedDevice}
+      value={selectedDeviceId || ''}
```
Comment (nicklera1) [inline src/components/sdk/DeviceSelection/DeviceInput.tsx:54]: maybe move this default at line 22 above as default prop valued ?

Comment (michhyun1) [reply src/components/sdk/DeviceSelection/DeviceInput.tsx:54]: yeah, will do

**Pull request merged**